go/ephemeral/device_ek.go

```diff
@@ -69,18 +69,22 @@ func publishNewDeviceEK(ctx context.Context, g *libkb.GlobalContext, merkleRoot
 	if err != nil {
 		// Let's try to get the max from the server
 		g.Log.CDebugf(ctx, "Error getting maxGeneration from storage")
-		generation, err = getServerMaxDeviceEK(ctx, g, merkleRoot)
+		generation, err = serverMaxDeviceEK(ctx, g, merkleRoot)
 		if err != nil {
 			return metadata, err
 		}
 	}
+	// This is our first generation
+	if generation < 0 {
+		generation = 0
+	}
 	generation++
```

Comment:
It might be cleaner to just fold this `++` into each branch above. Same with the other occurrences of `generation++` in this PR. I think they made more sense back when there were retries?

Reply:
for devices we still do the retry since we retry on local storage for `getAll`

Reply:
i'll change it for user/team stuff though